Modules/posixmodule.c

```diff
@@ -4389,6 +4390,40 @@ os__getvolumepathname_impl(PyObject *module, path_t *path)
     return result;
 }
 
+
+/*[clinic input]
+os._path_splitroot
+
+    path: path_t
+
+Removes everything after the root on Win32.
+[clinic start generated code]*/
+
+static PyObject *
+os__path_splitroot_impl(PyObject *module, path_t *path)
+/*[clinic end generated code: output=ab7f1a88b654581c input=dc93b1d3984cffb6]*/
+{
+    wchar_t *end;
+    PyObject *result = NULL;
+    HRESULT ret;
+
+    Py_BEGIN_ALLOW_THREADS
+    ret = PathCchSkipRoot(path->wide, &end);
```

Comment:
If you copy `path->wide` with slashes replaced with backslashes, then the offset of `end` in the copy can be used to split `path->wide` with the original mixed slashes.

Reply:
Yeah, I was trying to avoid the copy (specifically, the wstrlen and dynamic allocation) in C, but I think it's the better approach.